e2e-tests/src/test/rewards.rs

```diff
@@ -311,3 +304,82 @@ pub fn force_new_era(config: &Config) -> anyhow::Result<()> {
 
     Ok(())
 }
+
+pub fn change_stake_and_force_new_era(config: &Config) -> anyhow::Result<()> {
+    const MAX_DIFFERENCE: f64 = 0.07;
+
+    let node = &config.node;
+    let accounts = get_validators_keys(config);
+    let sender = accounts.first().expect("Using default accounts").to_owned();
+    let connection = SignedConnection::new(node, sender);
+
+    let sudo = get_sudo_key(config);
+    let root_connection = RootConnection::new(node, sudo);
+
+    let (reserved_members, non_reserved_members) = get_member_accounts(config);
+
+    wait_for_full_era_completion(&connection)?;
+
+    let start_era = get_current_era(&connection);
+    let start_session = get_current_session(&connection);
+    info!("Start | era: {}, session: {}", start_era, start_session);
+
+    validators_bond_extra_stakes(
+        config,
+        [
+            7 * MIN_VALIDATOR_BOND,
+            2 * MIN_VALIDATOR_BOND,
+            11 * MIN_VALIDATOR_BOND,
+            0,
+            4 * MIN_VALIDATOR_BOND,
+        ]
+        .to_vec(),
+    );
+
+    staking_force_new_era(&root_connection, XtStatus::Finalized);
+
+    wait_for_session(&connection, start_session + 2)?;
+    let current_era = get_current_era(&connection);
+    let current_session = get_current_session(&connection);
+    info!(
+        "After ForceNewEra | era: {}, session: {}",
+        current_era, current_session
+    );
+
+    // Once a new era is forced in session k, the new era does not come into effect until session
+    // k + 2; we test points:
+    // 1) immediately following the call in session k,
+    // 2) in the interim session k + 1,
+    // 3) in session k + 2, the first session of the new era.
+    for idx in 0..3 {
```

Comment:
Can this for loop be extracted to some common place, as the above test `force_new_era` seems to use it as well?